csrc/buffer.h

```diff
@@ -598,6 +598,12 @@ class JByteArrayCritical {
     ~JByteArrayCritical();
     unsigned char* get();
 
+#ifdef HAVE_CPP11
+    // deleting copy constructor and copy assignment to satisfy rule of three
+    JByteArrayCritical(const JByteArrayCritical&) = delete;
+    JByteArrayCritical& operator=(const JByteArrayCritical&) = delete;
+#endif
```

Comment:
nit: We could add an #else here to handle the RHEL5 case by using empty/private overrides. I'd prefer fewer LOC than handling EOL/Obsolete OS's though.

Reply:
What do you prefer? Removing them just on C++11 compilers should catch most of the potential mis-uses of the classes. But there may be a corner case where a pre-C++11 compiler does a copy while a C++11 compiler does not (e.g., because some `#ifdef` somewhere else makes the pre-C++11 code slightly different).

Reply:
I'd avoid adding code to handle RHEL5 here. Forget I ever brought it up.